SalesforceCommerceCloud/pwa-kit
# [Storefront Preview 🔍 ] Add "onContextChange" Property to StorefrontPreview Component

# Description

This is the companion PR for [#466](https://git.soma.salesforce.com/cc-mobify/runtime-admin-ui/pull/446) in the Runtime-Admin project.

The goal in this PR is the expose the context changed message event in the form of a handler. This will allow StorefrontPreview implementers to define this handler and take the context and do what they want with it. The use case we are looking at is to integrate 3rd party CMS's like Amplience. 

# Types of Changes


# Changes

- Add `onContextChange` prop to StorefrontPreview component.
- Move custom prop validator to utilities

# How to Test-Drive This PR

- Visit this [preview link](https://runtime-admin-preview.mobify-storefront.com/kyle-preview-demo/p/test-ben/preview)
- The page will load, and after 7.5s you should see the data reload navigation happen. 
- This happens after 7.5 seconds because the storefront defined a 7.5 second sleep in the form of an async onContextChange handler.
- You can repeat this by clicking the "Preview" button, the button will immediately grey out, but the page will only load after 7.5s 

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
